### ISSUE TITLE
 Spawn at structure option

### ISSUE THREAD

**Issue by LucyNoel** [Team-MCToolkit/MCToolkit#177]
Describe the feature

An option where you can pick the specific structure where the entity or block spawns.

How would you implement it?

There would be a spawn at custom structure option, and a spawn at a Minecraft structure option on the spawning or generation page.
Along with  the option to pick the specific custom structure or the specific Minecraft structure it spawns at.

Additional context

For example you could choose for it to spawn in a desert village if you chose the Minecraft structure option.

But if you chose the custom structure option you can pick one of your structures for it to spawn at.

**Comment by Trolmaso** [comment]
So for example, you can make a custom “villager” spawn in a village?

**Comment by LucyNoel** [comment]
Yes, but also it could spawn in a different structure. Like any custom one or any structure that was already made in Minecraft. Also it doesn't have to just  be a villager, it could be any entity!

**Comment by northwesttreesgaming** [comment]
Internal spawns may not be possible to control, however you can control the first spawn block with vanilla tag "valid_spawn" and select the block you want to allow players to spawn on. For example if you add the block "grass_path" to the tag in your mod then you will allow players to respawn/spawn on grass_paths.

Other than that the devs will have to reply after they see if its possible to do or not in regard of making structures spawn locations, some things are limited to what can be done with Forge so there is no guarantee it will be possible or not.

**Comment by Goldorion** [comment]
I don't know if we can implement that, but Jigsaw blocks allow to spawn a structure or an entity directly inside the structure. However, there was a bug with them, so we couldn't use them in mods, but I don't know if it was fixed.
